sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateInitialStateSuite.scala

```diff
@@ -83,6 +98,59 @@ abstract class StatefulProcessorWithInitialStateTestClass[V]
   }
 }
 
+/**
+ * Stateful processor that will take a union dataframe output from state data source reader,
+ * with flattened state data source rows.
+ */
+class InitialStatefulProcessorWithStateDataSource
+  extends StatefulProcessorWithInitialStateTestClass[UnionInitialStateRow] {
+  override def handleInitialState(
+    key: String, initialState: UnionInitialStateRow, timerValues: TimerValues): Unit = {
+    val stateVar = {
+      if (initialState.value.isDefined) "value"
```

Comment:
nit: lets use the conditions with braces ?

`if () {

} else if () {

} else {

}
```

Reply:
Done!